src/jetstream/middleware.go

```diff
@@ -39,7 +41,7 @@ const APIKeyAuthScheme = "Bearer"
 func handleSessionError(config api.PortalConfig, c echo.Context, err error, doNotLog bool, msg string) error {
 	log.Debug("handleSessionError")
 
-	if strings.Contains(err.Error(), "dial tcp") {
+	if errors.Is(err, errorz.ErrDialTcp) {
```

Comment:
As discussed this is not quite the right test, as go will compare the actual error objects for equality here.
Because the net.OpError that we are expecting here (the test was very narrow here, only expecting the error string to contain 'dial tcp') is a very complex structure (https://pkg.go.dev/net#OpError) it will never be equal to the freshly constructed errorz.ErrDialTcp.
Here's an example that might help you: 
https://play.golang.com/p/75xzJyyz3S9